### PR TITLE
Enable nonblocking operation

### PR DESCRIPTION
**Pull request by andreasley**
This PR is part of a series to enable nonblocking operation in Vapor and solve https://github.com/vapor/engine/issues/66

This PR depends on https://github.com/vapor/socks/pull/86.

~~If this PR is accepted, Vapor needs to be updated with https://github.com/vapor/vapor/pull/690 to work again.~~

### CODE REVIEW

**Comment by LoganWright** [comment]
@andreasley is there an appropriate set of defaults for watchable that we could implement to properly deprecate w/o breaking stream conformance api?

**Comment by andreasley** [comment]
@LoganWright Absolutely; we can just add the following extension:

```swift
extension Stream {
    public func startWatching(on queue: DispatchQueue, handler: @escaping () -> ()) throws {
        throw StreamError.unsupported
    }
    
    public func stopWatching() throws {
        throw StreamError.unsupported
    }
}
```

This way, the placeholder implementations in other `Stream` classes become redundant and can be removed. 

I have added a commit with these changes. Should have done it like this from the start; not sure what I was thinking. :)

File: Sources/HTTP/Server/ServerError.swift [hunk 1]
```diff
@@ -6,4 +6,5 @@ public enum ServerError: Swift.Error {
     case respond(Swift.Error)
     case dispatch(Swift.Error)
     case unknown(Swift.Error)
+    case notSupported
```
Comment (LoganWright) [inline Sources/HTTP/Server/ServerError.swift]: We'll have to pull this out unfortunately. We can throw a `respond` error and have a `notSupported` error somewhere else for API preservation.

Comment (andreasley) [inline Sources/HTTP/Server/ServerError.swift]: Fine by me, but `.respond` still requires a Swift.Error as parameter. Are there any generic ones available?

**Pull request merged**